cmd/goal/application.go

```diff
@@ -414,12 +414,14 @@ func mustParseProgArgs() (approval []byte, clear []byte) {
 	if approvalProgFile != "" {
 		approval = assembleFile(approvalProgFile, false)
 	} else {
+		// TODO: this is unreachable code bc of reportErrorf check above!!!
```

Comment:
why is it unreachable? `approvalProgFile == ""` assumes `approvalProgRawFile != ""`

Reply:
Thanks for prompting with this question. Looking a little closer at the logic, at line 407 above we have:

```go
if (approvalProgFile == "") == (approvalProgRawFile == "") {
		reportErrorf(errorApprovProgArgsRequired)
	}
```

So in the case that _BOTH_ `approvalProgFile` and `approvalProgRawFile` are empty we never get to line 418. However, there is the case that `approvalProgFile` is empty, but `approvalProgRawFile` is _non-empty_, in which case we can still reach here this code and in fact it makes sense to use the raw file. I haven't looked closely enough at the relationship between these two files to see if that is an actual possibility.

I will conduct a bit more research to see the relationship.

Reply:
Ok, I see now that these are completely independent variables/command line flags with no dependencies. So I will remove my false commentary.